apps/glusterfs/brick_allocate.go

```diff
@@ -204,6 +204,8 @@ func tryAllocateBrickOnDevice(
 		logger.Debug(
 			"Unable to place a brick of size %v & factor %v on device %v",
 			brickSize, snapFactor, device.Info.Id)
+	} else {
+		brick.SubType = NormalSubType
```

Comment:
This seems to be an initialization to a default value. Why initialize here instead of in `brick_entry.go:NewBrickEntry()`?

Reply:
We could set the sub type in NewBrickEntry but the only code that "knows" the correct subtype is the placer. Normal is not the default, Unknown sub type is the default value. 

The lifecycle of the new field goes something like:
* All old entries pulled from the db will get an "unknown" sub type
* Placers that create a new brick entry set the correct sub type
* The new sub type values will get stored in the db
* The create brick code path uses the value to determine the formatting for new brick fs

This approach was designed not to need a specific upgrade function. However, adding a flag to indicate that this db can contain subtypes seems reasonable.

Reply:
ok. well then!